app/modules/asset_groups/parameters.py

```diff
@@ -241,9 +242,27 @@ def _get_encounter_data(cls, obj, state):
     @classmethod
     def add(cls, obj, field, value, state):
 
-        # Annotations are special in that it can only be added, not replaced.
-        if field == 'annotations':
-            encounter_uuid, encounter_metadata = cls._get_encounter_data(obj, state)
+        # For everything, Add and replace are the same operation so reuse the one method
+        return cls.replace(obj, field, value, state)
+
+    @classmethod
+    def replace(cls, obj, field, value, state):
+
+        ret_val = True
+        encounter_uuid, encounter_metadata = cls._get_encounter_data(obj, state)
+
+        if field == 'ownerEmail':
+            AssetGroupMetadata.validate_owner_email(value, f'Encounter {encounter_uuid}')
+            encounter_metadata[field] = value
+        elif field == 'owner':
+            from app.modules.users.models import User
+
+            user = User.query.get(value)
+            if user and user.is_researcher:
+                encounter_metadata['ownerEmail'] = user.email
+
+        elif field == 'annotations':
```

Comment:
This was what we agreed but I'm not sure that replace adding annotations to teh encounter makes sense.

Reply:
It does seem a little weird. Follows the pattern that we have used before where add calls replace though. I'm happy to talk about it again, don't want to add confusion.